File: aspnet-core/src/RMALMS.Application/CertificationTemplates/Dto/CertificationTemplateDto.cs
using Abp.Application.Services.Dto;
using Abp.AutoMapper;
using RMALMS.Entities;
using System;
using RMALMS.Anotations;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using Microsoft.AspNetCore.Http;

namespace RMALMS.CertificationTemplates.Dto
{
    [AutoMapTo(typeof(CourseCertificationTemplate))]
    public class CertificationTemplateDto : EntityDto<Guid>
    {
        public string Name { get; set; }
        public Guid CourseId { get; set; }
        public bool IsActive { get; set; }
        public string Background { get; set; }
        public string Content { get; set; }
        public CertificationType CertificationType { get; set; }
        public IFormFile File { get; set; }
        public CertificationOrientation Orientation { get; set; }


    }
}

File: angular/src/app/models/certificationtemplate-dto.ts


export interface ICertificationTemplateDto {
    id: string;
    isActive: boolean;
    courseId: string;
    background: string;
    name: string;
    content: string;
    certificationType: number;
    orientation: number
}

export class CertificationTemplateDto implements ICertificationTemplateDto {
    id: string;
    isActive: boolean;
    courseId: string;
    background: string;
    name: string;
    content: string;
    certificationType: number;
    isEdit: boolean;
    file: File;
    imgBase64Value: string;
    orientation: number;
    isView: boolean;


}

export interface ICreateCertificationTemplateDto {
    isActive: boolean;
    courseId: string;
    name: string;
    content: string;
    certificationType: number;
    orientation: number

}

export class CreateCertificationTemplateDto implements ICreateCertificationTemplateDto {
    isActive: boolean;
    courseId: string;
    name: string;
    content: string;
    certificationType: number;
    file: File;
    isEdit: boolean;
    orientation: number
}

export interface IResultDto {
    result: CertificationTemplateDto[];
}
